document_page_access_group/models/document_page.py

```diff
@@ -1,10 +1,19 @@
 # Copyright 2022 Manuel Regidor <manuel.regidor@sygel.es>
+# Copyright 2024 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
 
 
 class DocumentPage(models.Model):
     _inherit = "document.page"
 
     groups_id = fields.Many2many(comodel_name="res.groups", string="Groups")
+    user_ids = fields.Many2many(comodel_name="res.users", string="Users")
+
+    @api.constrains("groups_id", "user_ids")
+    def check_document_page_groups_users(self):
```

Comment:
This constrains is not being applied, check this example:

1. Add one role to one document page
![image](https://github.com/user-attachments/assets/16589517-d82c-4f2c-897e-539acbdf06ab)

2. Add one groups and save:

![image](https://github.com/user-attachments/assets/4ab44df6-7ce1-44ac-b4dc-f247878e05d0)

3. Roles now gets invisible
![image](https://github.com/user-attachments/assets/6d7e6064-0414-478c-9b8a-b9f4494a027c)

4. Remove group and then you will see that role is still applied:

![image](https://github.com/user-attachments/assets/9b4139c0-d943-4fd8-81cc-fb1f36be86a4)

Reply:
I think it is important to clarify that constrains is related to **groups** and **users** and is something that cannot be achieved by UX. If we perform a `write()` of a page set groups and users the error will be displayed (this is what we want to avoid, the data inconsistency).